src/matrix_content_scanner/scanner/scanner.py

```diff
@@ -434,6 +488,30 @@ def _decrypt_file(self, body: bytes, metadata: JsonDict) -> bytes:
                 info=str(e),
             )
 
+    async def write_multipart_to_disk(self, multipart: BodyPartReader) -> str:
+        """
+        Writes a multipart file body to the store directory.
+
+        Returns:
+            The full file path to the file.
+        """
+        filename = str(uuid.uuid4())
+        # Figure out the full absolute path for this file.
+        full_path = self._store_directory.joinpath(filename).resolve()
+        logger.info("Writing multipart file to %s", full_path)
+
+        # Create any directory we need.
+        os.makedirs(full_path.parent, exist_ok=True)
+
+        with open(full_path, "wb") as fp:
+            while True:
+                chunk = await multipart.read_chunk()
+                if not chunk:
+                    break
+                fp.write(chunk)
```

Comment:
this would probably be *better* if it was async, but seems that we might have to pull in a library for this. I also guess you aren't the first to do it this way.
If you were interested though, https://pypi.org/project/aiofile/ looks reasonable